sqlmesh/core/schema_diff.py

```diff
@@ -456,6 +458,9 @@ def _alter_operation(
                     root_struct,
                 )
             if new_type.this == current_type.this == exp.DataType.Type.ARRAY:
+                # Some engines (i.e. Snowflake) don't support defining types on arrays
+                if not new_type.expressions or not current_type.expressions:
```

Comment:
what if something is untyped and then it becomes typed?

Reply:
There is no known way to do this today. The only untyped example must remain untyped.